internal/federation/server.go

```diff
@@ -94,7 +95,8 @@ func NewServer(t *testing.T, deployment *docker.Deployment, opts ...func(*Server
 	})
 	srv.mux.NotFoundHandler = http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		if srv.UnexpectedRequestsAreErrors {
-			t.Errorf("Server.UnexpectedRequestsAreErrors=true received unexpected request to server: %s %s", req.Method, req.URL.Path)
+			body, _ := ioutil.ReadAll(req.Body)
+			t.Errorf("Server.UnexpectedRequestsAreErrors=true received unexpected request to server: %s %s\n%s", req.Method, req.URL.Path, string(body))
```

Comment:
I'm assuming `string(io.ReadCloser)` is safe and isn't going to fall over with e.g. a sequence of bytes that's not e.g. a valid UTF-8 byte stream?

Reply:
(I did try looking at the go docs but couldn't see where `string(io.ReadCloser)` would be described or implemented)

Reply:
(given we're failing the test anyway at this point, I don't think it matters too much.)